### PR TITLE
feat: AdminWidget Enable/Disable/RemovePasskey and editable support

### PR DESCRIPTION
**Pull request by tomerlichtash**
## Related Issues

Related screens https://github.com/descope/content/pull/417

## Related PRs

| branch       | PR         |
| ------------ | ---------- |
| service a PR | Link to PR |
| service b PR | Link to PR |

## Description

A few sentences describing the overall goals of the pull request's commits.

## Must

- [ ] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

**Review by nirgur — COMMENTED** [review]
Looks good
Added some comments

File: packages/user-management-widget/src/lib/widget/mixins/initMixin/initMixin.ts [hunk 1]
```diff
@@ -8,17 +8,23 @@ import { initFilterUsersInputMixin } from './initComponentsMixins/initFilterUser
...
 import { initUsersTableMixin } from './initComponentsMixins/initUsersTableMixin';
 import { initEditUserButtonMixin } from './initComponentsMixins/initEditUsersButtonMixin';
+import { initEnableUserButtonMixin } from './initComponentsMixins/initEnableUserButtonMixin';
+import { initDisableUserButtonMixin } from './initComponentsMixins/initDisableUserButtonMixin';
+import { initRemovePasskeyButtonMixin } from './initComponentsMixins/initRemovePasskeyButtonMixin';
 
 export const initMixin = createSingletonMixin(
   <T extends CustomElementConstructor>(superclass: T) =>
-    // @ts-ignore
+    /* @ts-ignore */
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/src/lib/widget/mixins/initMixin/initMixin.ts:17]: Looks like an [open issue in Typescript](https://github.com/microsoft/TypeScript/issues/34933). Let's leave `ts-ignore` for now then?

File: packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initDisableUserButtonMixin.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        this.disableButton.disable();
+        this.disableButton.onClick(() => {
+          this.disableUserModal.open();
+        });
+      }
+
+      #onIsUserSelectedUpdate = withMemCache(
+        (isEnabled: ReturnType<typeof getIsSelectedUsersEnabled>) => {
+          if (isEnabled) {
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initDisableUserButtonMixin.ts:37]: `editable` should be handled for all actions; what did we decide on the desired behavior? Keep the buttons disabled? Open the edit modal with disabled inputs?

I think it'd be be best if we support this feature in a separate PR, wdyt?

**Comment by tomerlichtash** [comment]
Added support for custom attributes in edit user modal. FYI @aviadl

File: packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initDisableUserButtonMixin.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nirgur) [reply packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initDisableUserButtonMixin.ts:37]: AFAIR, we said that all buttons should be disabled except delete user
Need to verify it with Aviad

File: packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initEditUserModalMixin.ts [hunk 3]
```diff
@@ -89,6 +104,12 @@ export const initEditUserModalMixin = createSingletonMixin(
           email: userDetails?.email,
           phone: formatPhoneNumber(userDetails?.phone),
           roles: userDetails?.roles,
+          ...Object.fromEntries(
+            Object.entries(userDetails.customAttributes).map(([key, val]) => [
+              `customAttributes.${key}`,
+              val,
+            ]),
+          ),
```
Comment (nirgur) [inline packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initEditUserModalMixin.ts:112]: Let's extract it into a `flatten` function

**Review by nirgur — COMMENTED** [review]
Looks good
I think we should make it a bit more generic

LMKWYT

File: packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initEditUserModalMixin.ts (hunk 3 above)
Comment (tomerlichtash) [reply packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initEditUserModalMixin.ts]: Dome

File: packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initEditUserModalMixin.ts [hunk 4]
```diff
@@ -12,6 +12,21 @@ import { TextFieldDriver } from '../../../drivers/TextFieldDriver';
...
+      const customKey = key.split('.')[1];
+      if (!acc['customAttributes']) {
+        acc['customAttributes'] = {};
+      }
+      acc['customAttributes'][customKey] = value;
+    } else {
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initEditUserModalMixin.ts]: Done

**Review by nirgur — APPROVED** [review]

**Pull request merged**